src/DIRAC/__init__.py

```diff
@@ -167,20 +169,105 @@ def _computeRootPath(rootPath):
 from DIRAC.ConfigurationSystem.Client.Config import gConfig
 
 
-def initialize():
-    """Load configuration.
+from DIRAC.Core.Security.Properties import SecurityProperty, UnevaluatedProperty
+from DIRAC.Core.Utilities import exceptions
+from DIRAC.FrameworkSystem.private.standardLogging.LogLevels import LogLevel
+
+
+def initialize(
+    *,
+    require_auth: bool = True,
+    security_expression: Union[None, SecurityProperty, UnevaluatedProperty] = None,
+    log_level: Optional[LogLevel] = None,
+    extra_config_files: Optional[list[os.PathLike]] = None,
+    extra_config: Optional[dict[str, Any]] = None,
+    host_credentials: Optional[tuple[os.PathLike, os.PathLike]] = None,
+) -> None:
+    """Prepare the global state so that DIRAC clients can be used
+
+    This method needs to be called before any DIRAC client is created to ensure
+    that the necessary global state is configured.
+
+    If initialization fails, :py:class:`~DIRAC.Core.Utilities.exceptions.DIRACInitError`
```

Comment:
```suggestion
    :raises DIRAC.Core.Utilities.exceptions.DIRACInitError: If initialization fails.
```
(Not entirely sure about the module path)